### DISCUSSION TITLE
How to subscribe to a vanilla store and trigger the current state on subscribe?

### DISCUSSION THREAD

**Discussion by MiniSuperDev** [pmndrs/zustand#2120] [Q&A]
I'm using a vanilla store in the backend, so I do this to emit the current state
```ts
const disposer = store.subscribe((state) => {
    emitter.next(state);
  });
  emitter.next(store.getState());
  return () => {
    disposer();
  };
```

There is a built in way to subscribe and emit the current state?

Thanks

**Comment by dai-shi** [comment]
Your code snippet looks good to me. It's concise and predictable, isn't it?
